platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/container/ContainerNode.java

```diff
@@ -274,10 +308,19 @@ public void onCompleted() {
          * {@inheritDoc}
          */
         @Override
+        @SuppressWarnings("ResultOfMethodCallIgnored") // ignoring the Empty answer from killImmediately
         public void killImmediately() {
             log.info("Killing node {} immediately...", selfId);
-            if (channel != null) {
-                channel.shutdownNow();
+            try {
+                // Mark the node as shutting down *before* sending the request to avoid race
+                // conditions with the stream observer receiving an error.
+                lifeCycle = SHUTDOWN;
+
+                final KillImmediatelyRequest request = KillImmediatelyRequest.getDefaultInstance();
+                // Unary call – will throw if server returns an error.
+                blockingStub.killImmediately(request);
+            } catch (final Exception e) {
+                log.error("Failed to kill node {} immediately", selfId, e);
```

Comment:
I think we should fail the test directly or ignore any exceptions and let the caller deal with it.